src/hardware/sblaster.cpp

```diff
@@ -3046,6 +3046,46 @@ static void per_tick_callback()
 	frames_added_this_tick -= total_frames;
 }
 
+void SBLASTER::AudioCallback(const int frames_requested)
+{
+	if (MIXER_FastForwardModeEnabled()) {
+		// Special case, normally only hit when using the fast-forward
+		// hotkey (Alt + F12). We need a very large buffer to compensate
+		// or it results in static.
+
+		// Mostly arbitrary but it works well in testing.
+		// The queue just needs to be large enough to hold the large
+		// frame requests we get in fast-forward mode. This value can be
+		// tweaked without much consequence if it ever becomes
+		// problematic.
```

Comment:
💯 comment for the next guy maintaining this

Reply:
This entire function was mostly copy + pasted from the templated function in `mixer.h` btw.  I might figure a way to share common code once this PR is stabilized.  In the mean time, it's easier to make small changes with this function extracted.

Reply:
I was about to say, it would be good to move this into some helper function and have the big explanatory comments in one place.